ganttproject/src/main/java/net/sourceforge/ganttproject/storage/Xlog.kt

```diff
@@ -43,9 +46,29 @@ data class XlogRecord(
  * by baseTxnId, and i-th transaction from the list is applied to the state produced by (i-1)th transaction from the
  * list.
  */
+@Serializable
 data class InputXlog(
   val baseTxnId: String,
   val userId: String,
   val projectRefid: String,
   val transactions: List<XlogRecord>
+)
+
+/**
+ * Response from the server, which contains the result of applying changes sent by the client.
+ *
+ * @param baseTxnId the transaction to which the changes were applied.
+ * @param committedTxnId the resulting transaction.
+ * @param committedTxnNum the number of committed transactions.
+ *
+ * TODO: Get rid of type. It's used for routing in `WebSocketClient::onMessage`.
+ * TODO: Provide transaction conflicts handling.
+ */
+@Serializable
+data class ServerCommitResponse(
+  val baseTxnId: String,
+  val committedTxnId: String,
+  val committedTxnNum: Int,
```

Comment:
`txnNum` is the number of transactions which were consequently applied to the base. `txnId` is the resulting transaction id.
The client can send messages `{ "baseId", [txn1, txn2] }`, `{"baseId", [txn1, txn2, txn3]}`, `txnNum` helps to understand which request succeeded.

Reply:
I doubt that it makes sense to send different messages with the same base ID. The log is supposed to be linear, and the client should not create two tails forked from the same base point.

If you meant that it would be a linear log, just split across a few chunks and txnNum establishes the order of those chunks then I suggest to keep the protocol more simple: if we have a few transactions sequentially applied to some base ID then we send the first one, wait until we get the response, get a new base ID, then send the second one with the new base ID

Reply:
> If you meant that it would be a linear log
Yes, that's what I meant. The order is established by local xlog id. `txnNum` just tells which record to send next.

Okay, so that we never send a list of transactions, just a single one?

Reply:
For now, let's send just transactions one by one and see how it goes.

Reply:
Done